agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -3848,77 +3848,47 @@ bool slave_thread::handle_autoconfiguration_renew(Socket *sd, ieee1905_1::CmduMe
 bool slave_thread::handle_autoconfiguration_wsc(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
 {
     LOG(DEBUG) << "Received AP_AUTOCONFIGURATION_WSC_MESSAGE";
+    // According to section 17.1.3 in the Multi-AP Specification version 1.0,
+    // if an AP-Autoconfiguration WSC message is sent by the Multi-AP Agent,
+    // it includes one AP Radio basic capabilities TLV and one WSC TLV (containing M1).
+    // If the the message is sent by the Multi-AP controller, it includes one
+    // AP radio Identifier TLV and one or more WSC TLVs (containing M2).
+    // Check if this is the case and this is an M1 message we sent to the controller
+    // which was just looped back (when we send the M1 to the local controller,
+    // it will be published back on the local bus because the destination is our
+    // AL-MAC, and the agent does listen to this CMDU message type).
+    // If this is a looped back M1 CMDU, we can treat is as handled successfully.
+    if (cmdu_rx.getClass<wfa_map::tlvApRadioBasicCapabilities>())
```

Comment:
Shouldn't we check that it's indeed our looped request?
What about non-local agents (repeaters) that send an autoconfig message to our controller?
The local agent won't receive the messages?

Reply:
It doesn't matter if its our looped request or an M1 CMDU from another agent, in both cases we are dropping the message.

Reply:
We do get the other agents WSC messages and the handling is and should be the same as looped back M1 CMDUs - we should ignore them. Checking that this is indeed our looped request is a much harder thing to do with ZMQ which is our current MSGLIB backend. Perhaps it will become easier once we switch to a different implementation, but for now, this is the only way we can do that.